viewing_party/party.py

```diff
@@ -1,23 +1,128 @@
 # ------------- WAVE 1 --------------------
 
+from multiprocessing.sharedctypes import Value
+from re import I
+
+from tests.test_constants import GENRE_1, MOVIE_TITLE_1, RATING_1
+
+
 def create_movie(title, genre, rating):
-    pass
+    
+    new_movie = {}
+    if title and genre and rating:
+        new_movie = {
+            "title" : title,
+            "genre" : genre,
+            "rating" : rating
+        }
+        return new_movie
+    return None
+
+def add_to_watched(user_data, movie):
+    user_data = {"watched" : [movie]}
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data = {"watchlist" :[movie]}
+    return user_data
+
+def watch_movie(user_data, title):
+    for movie in user_data["watchlist"]:
+        if movie["title"] == title:
+            user_data["watched"].append(movie)
+            user_data["watchlist"].remove(movie)
+    return user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    average_rating = 0 
+    sum_list = []
+    for movie in user_data["watched"]:
+        if movie["rating"]:
+            sum_list.append(movie["rating"])
+        average_rating = sum(sum_list)/len(user_data["watched"])
+    return average_rating
 
+    
+def get_most_watched_genre(user_data):
+    genre_list =[]
+    for movie in user_data["watched"]:
+        if movie["genre"]:
+            genre_list.append(movie["genre"])
+            most_popular = max(set(genre_list), key = genre_list.count)
+        return most_popular

```

Comment:
Careful here! Let's look at the logic happening. For each iteration of the loop, we are calling the `max` function, as well as the `count` function. 

Then on line 55, what is happening? It's *inside* the loop, so what will a return do to a for loop? It will hit this line and stop before the loop is actually ready to be done.

I think we should move these two lines out of the for loop, so that they execute once for the for loop is done adding all the genres.

The reason this passed the test is that, looking at the test data, the first genre *was* the correct answer. So it happened to work with your code, even though the logic itself would likely fail with other test data.